packages/core/src/extensions/Blocks/api/blockTypes.ts

```diff
@@ -73,8 +73,21 @@ export type BlockConfig<
   containsInlineContent: ContainsInlineContent;
   parse?: (element: HTMLElement) => Props<PSchema>;
   render: (
-    block: Block<BSchema>,
-    editor: BlockNoteEditor<BSchema>
+    /**
+     * The custom block to render
+     */
+    block: SpecificBlock<
```

Comment:
Could this be simplified to
```
block: Block<{ [k in Type]: BlockSpec<Type, PSchema> }>
```
?

Reply:
Also as a side note - right now you could, if you wanted to, update `block` to a different type. Not relevant for now, but will be when we need to sort out cleaning up listeners, since I think that a listener for a custom block should be removed if the block changes type.

Reply:
I don't think it can, because the `children` of the block should be able to be of other types

Reply:
Ah, good point, missed that